### PR TITLE
[Merged by Bors] - feat: port Analysis.NormedSpace.OperatorNorm

### PR DESCRIPTION
**Pull request by kim-em**
---

This is the pushout of https://github.com/leanprover-community/mathlib4/pull/3708 and the reenableeta branch https://github.com/leanprover-community/mathlib4/pull/3414, to verify that even though #3708 is taking over 6 hours in CI, it is no problem with reenableeta.

- [x] depends on: #3414

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3414
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3414~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by jcommelin** [comment]
Can we rewrite the history on this PR, to make it easier to review? I would like to get the diff between mathport output and the HEAD of the PR.

**Comment by Parcly-Taxel** [comment]
I'm gonna try…

**Comment by Ruben-VandeVelde** [comment]
> Can we rewrite the history on this PR, to make it easier to review? I would like to get the diff between mathport output and the HEAD of the PR.

Done.

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean [hunk 1]
```diff
@@ -464,22 +453,30 @@ theorem op_norm_comp_le (f : E →SL[σ₁₂] F) : ‖h.comp f‖ ≤ ‖h‖ *
...
-omit σ₁₃
-
 /-- Continuous linear maps form a seminormed ring with respect to the operator norm. -/
 instance toSemiNormedRing : SeminormedRing (E →L[𝕜] E) :=
   { ContinuousLinearMap.toSeminormedAddCommGroup, ContinuousLinearMap.ring with
     norm_mul := fun f g => op_norm_comp_le f g }
 #align continuous_linear_map.to_semi_normed_ring ContinuousLinearMap.toSemiNormedRing
 
+-- Porting FIXME: replacing `(algebra : Algebra 𝕜 (E →L[𝕜] E))` with
+-- just `algebra` below causes a massive timeout.
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/OperatorNorm.lean:472]: Is this still true?

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean [hunk 2]
```diff
@@ -807,9 +808,11 @@ theorem flip_smul (c : 𝕜₃) (f : E →SL[σ₁₃] F →SL[σ₂₃] G) : (c
 
 variable (E F G σ₁₃ σ₂₃)
 
+set_option maxHeartbeats 300000 in
+set_option synthInstance.maxHeartbeats 25000 in
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/OperatorNorm.lean:812]: Is this still needed?

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean [hunk 3]
```diff
@@ -822,21 +825,24 @@ def flipₗᵢ' : (E →SL[σ₁₃] F →SL[σ₂₃] G) ≃ₗᵢ[𝕜₃] F 
...
 
+set_option synthInstance.maxHeartbeats 75000 in
 @[simp]
 theorem coe_flipₗᵢ' : ⇑(flipₗᵢ' E F G σ₂₃ σ₁₃) = flip :=
   rfl
 #align continuous_linear_map.coe_flipₗᵢ' ContinuousLinearMap.coe_flipₗᵢ'
 
 variable (𝕜 E Fₗ Gₗ)
 
+set_option maxHeartbeats 250000 in
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/OperatorNorm.lean:842]: Is this still needed?

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean [hunk 4]
```diff
@@ -987,7 +1006,8 @@ variable (M₁ : Type u₁) [SeminormedAddCommGroup M₁] [NormedSpace 𝕜 M₁
 
 variable {Eₗ} (𝕜)
 
-/-- `continuous_linear_map.prod_map` as a continuous linear map. -/
+set_option maxHeartbeats 750000 in
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/OperatorNorm.lean:1009]: Is this still needed?

**Review by jcommelin — COMMENTED** [review]
Thanks so much for cleaning up the git history! And for all the valiant work on this monster PR!

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 1 above)
Comment (Parcly-Taxel) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:472]: It's either this or `set_option maxHeartbeats 2000000` (2 million). 1 million is not enough.

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 3 above)
Comment (Parcly-Taxel) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:842]: See above.

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 4 above)
Comment (Parcly-Taxel) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:1009]: See above.

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 2 above)
Comment (jcommelin) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:812]: Thanks for confirming!

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 1 above)
Comment (jcommelin) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:472]: Thanks for confirming!

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**